provider/specs/client.md

```diff
@@ -0,0 +1,82 @@
+3. If polling is enabled, the function should start the polling loop *(See [polling section](#polling))*.
+
+## Evaluation
+The evaluation should not do any remote API calls.
+
+When calling an evaluation function the provider should check if the associated type is supported, by checking the key `capabilities.flagEvaluation.unsupportedTypes` from the configuration endpoint.
+- If the type is unsupported, we should exit early and directly return an error.
+- If the type is supported, we should retrieve the flag from the local cache of the flags evaluation.
+  - If you are not able to find the flag, you should return an FlagNotFound error.
+  - If the remote evaluation for this flag has return an error, you should map the error in provider and return the associated error.
+  - If the value retrieve from the cache has a different type than the one expected you should return a TypeMismatch error.
+  - If the cached evaluation is in success you should return the evaluation response.
+
+## Polling
+The polling system will make a POST request periodically tp the `/ofrep/v1/evaluate/flags` endpoint to check if there is a change in the flags evaluation to be able to store it.
+
+If we have a `ETag` available we should always add the header `If-None-Match` with the `ETag` value.
+- If the cache is still up-to-date we will receive a `304` telling us that the nothing has changed on the flag management system side.
+- If the cache is outdated we will receive a `200` with the new values of all the flags. In that situation we should:
+  1. Replace the actual local cache of flags evaluations.
+  2. Store the new `ETag` value for the future call.
+
+
+## Annexe 1
+**`/ofrep/v1/configuration` description:**
+The endpoint will return a list of configuration for the provider:
+- `name`: Name of the flag management system, it should be used in the `metadata.name` name (ex: `OFREP Web Provider ${metadata.name}`).
+- `capabilities`: List of capabilities of the flag management system and their associated configuration. *Check the [capabilities section](#capabilities) for the details of each capability.*
+
+### Capabilities
+In this section we will describe each capabilities and describe their default value and usage.
+
+#### Cache Invalidation
+The capability `cacheInvalidation` describe how the mechanism of cache invalidation is working.
+
+##### Polling
+`polling`: define how the provider should do the polling
+
+- `enabled`: if `true` the provider should poll the `/ofrep/v1/evaluate/flags` regularly to check if any flag evaluation has changed in the flag management system.
+- `minPollingInterval`: define the minimum polling interval acceptable by the flag management system. If for any reason the `pollInteral` provided in the constructor is lower than this `minPollingInterval` we should default on this value.
+
+If the key `polling` is not available, the provider should use those default values:
+```json
+{
+    "enabled": true,
+    "minPollingInterval": 0 
+}
```

Comment:
This is best represented as a table. Besides, I think we should define a non-zero `minPollingInterval` default value

Reply:
What would be an acceptable value for you? I was thinking at something like 60 seconds.